src/JsonSchema/Type/ElaoEnumType.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\JsonSchema\Type;
+
+use ApiPlatform\Core\JsonSchema\Schema;
+use ApiPlatform\Core\JsonSchema\TypeFactoryInterface;
+use Elao\Enum\EnumInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+class ElaoEnumType implements TypeFactoryInterface
+{
+    /**
+     * @var TypeFactoryInterface
+     */
+    private $decorated;
+
+    public function __construct(TypeFactoryInterface $decorated)
+    {
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getType(Type $type, string $format = 'json', ?bool $readableLink = null, ?array $serializerContext = null, Schema $schema = null): array
+    {
+        if (!is_a($enumClass = $type->getClassName(), EnumInterface::class, true)) {
+            return $this->decorated->getType($type, $format, $readableLink, $serializerContext, $schema);
+        }
+
+        $values = $enumClass::values();
+
+        return [
+            'type' => 'string',
+            'enum' => $values,
+            'example' => reset($values),
```

Comment:
I'm not sure to get why the `reset` here, it should work without it right?

Reply:
Because $values is an array and example takes a string afaik. Its just to get the first value as example.

Reply:
Indeed, or just ` $values[0]` if that works.

Reply:
I thought it was not number-indexed but i checked in my project and it is. It changed it to $values[0].